src/generators/utils/parse.ts

```diff
@@ -1,9 +1,22 @@
+import {isURLMatched, isFullyQualifiedDomain} from '../../utils/url';
 import {parseArray} from '../../utils/text';
 
 interface SiteProps {
     url: string[];
 }
 
+interface SitePropsOffsetRecord {
```

Comment:
My algorithm creates an instance of `SitePropsOffsetRecord` for every site prop. That is not very efficient, but very easy to read. I would prefer to sacrifice some readability and store these offsets in serialized form in a flat number array or even a string (hichever is more compact). Thoughts?

Reply:
Yes, this is close to what I was thinking. I want to improve on this design just a bit by storing the length of delimiters (to avoid having to strip them out twice) and encoding the flat number array into a string (because base 10 numbers are not efficient).

Reply:
I switched to a flat array with even items representing start and odd representing end. This way, if we need offsets of item with index `i`, its start is at `2*i` and end is at `2*i+1`. There is very little memory savings in "shingled" offsets and a bit more code complexity. Most lot of memory usage comes from domain URL strings themselves (keys in index).

Reply:
Sure things.